apluslms_roman/cli.py

```diff
@@ -379,6 +413,142 @@ def config_print_action(context):
+            raise RomanConfigError
+        exit(1, "{} already exists in {}.\nNo new Roman config created.".format(
+            basename(config_path), cwd)
+        )
+    except RomanConfigError:
+        filename = DEFAULT_NAME + ".yml"
+        config = LEGACY_CONFIG if is_legacy_configuration(cwd) else DEFAULT_CONFIG
+        filepath = path_join(cwd, filename)
+        with open(filepath, 'w') as outfile:
+            yaml_dump(config, outfile)
+        print("{} successfully created in {}".format(filename, cwd))
+
+def set_args(context):
+    if context.args.user == context.args.project:
+        raise Exception("Choose either user settings or project settings")
+    if context.args.user:
+        document = context.settings
+    else:
+        document = context.config
+
+    schema = document.validator.schema
+    for val in context.args.values:
+        try:
+            key, val = val.split('=')
+            parts = key.split(".")
+            val_type = schema
+            try:
+                for part in parts:
+                    val_type = val_type["properties"][part]
+                val_type = val_type["type"]
+                if val_type == "integer":
+                    if val.isdigit():
+                        val = int(val)
+                    else:
+                        raise TypeError("'" + key + "' should be an integer.")
+                elif val_type == "boolean":
+                    if val.lower == "true" or val == "1":
+                        val = True
+                    elif val.lower == "false" or val == "0":
+                        val = False
+                    else:
+                        raise TypeError("'" + key + "' should be a boolean.")
+            except KeyError:
+                raise KeyError("'" + key + "' is not a key found in the schema.")
+            document.mlset(key, val)
+        except ValueError:
+            raise ValueError("Give values in format 'key=val'.")
+    document.validate()
+    document.save()
+
+def del_args(context):
+    if context.args.user == context.args.project:
+        raise Exception("Choose either user settings or project settings")
+    if context.args.user:
+        document = context.settings
+    else:
+        document = context.config
+    for val in context.args.keys:
+        document.mldel(val)
+    document.validate()
+    document.save()
+
+def add_step(context):
+    args = context.args
+    env = args.env
+    if env:
+        try:
+            env = {a: b for a, b in [s.split('=') for s in env]}
+        except ValueError:
+            raise ValueError("env is a dict, so values need to be in key=val format, e.g. a=1 b=2")
+    step = {
+        "img": args.img,
+        "cmd": args.cmd,
+        "mnt": args.mnt,
+        "env": env,
+        "name": args.name
+    }
+    nulls = [k for k in step if not step[k]]
+    for k in nulls:
+        del step[k]
+    for k in step:
+        if k != "env":
+            step[k] = step[k][0]
```

Comment:
Is this here because there is `nargs=1` instead of `nargs=None`?

Reply:
I can't remember anything about this :') well, shouldn't be too hard to find out...

Reply:
okay yeah I must've assumed nargs=1 would mean that there is one argument. which in my opinion is a completely reasonable assumption...?

Reply:
nargs=1 and nargs=None both mean "there is only one argument", but the value returned is in different format. IF I read the code correctly, nargs=None produces a single value and nargs=1 produces a list (else branch in following code).

ref: https://github.com/python/cpython/blob/master/Lib/argparse.py#L2375